src/CoolPropLib.cpp

```diff
@@ -419,6 +419,29 @@ EXPORT_CODE long CONVENTION AbstractState_factory(const char* backend, const cha
 	}
     return -1;
 }
+EXPORT_CODE void CONVENTION AbstractState_fluid_names(const long handle, char* fluids, long *errcode, char *message_buffer, const long buffer_length)
+{
+    *errcode = 0;
+
+    try{
+        shared_ptr<CoolProp::AbstractState> &AS = handle_manager.get(handle);
+        std::vector<std::string> _fluids = AS->fluid_names();
```

Comment:
Use the string join method: https://github.com/CoolProp/CoolProp/blob/339a7ac4977bace36947ef6608b68bc258afdc2c/include/CPstrings.h#L113